lib/matplotlib/cm.py

```diff
@@ -735,5 +735,5 @@ def _ensure_cmap(cmap):
     cmap_name = cmap if cmap is not None else mpl.rcParams["image.cmap"]
     # use check_in_list to ensure type stability of the exception raised by
     # the internal usage of this (ValueError vs KeyError)
-    _api.check_in_list(sorted(_colormaps), cmap=cmap_name)
```

Comment:
This matters for the printed error message when the colormap is invalid.  (We could instead ensure that _colormaps is always sorted by sorting it whenever a new item is added, though.)

Reply:
I see. We could ensure the ordering of `_colormaps` by adding `c._cmaps = {k: c._cmaps[k] for k in sorted(c._cmaps)}` at the end of `ColormapRegistry.register`. Under the assumption that number of colormaps added is reasonable, this will not have any other negative performance impact.

Another option is to replace the code with
```
if not cmap_name in _colormaps:
    _api.check_in_list(sorted(_colormaps), cmap=cmap_name)
```

Reply:
Making sure that `__iter__` on `ColormapRegistry` is always sorted is probably the best option (rather than mucking with dictionary order we can keep a `self._sorted_keys = sorted(self._cmap)` attribute and iterate over that in `__iter__`.

Reply:
but the `if not in...` is also fine (as I see you already pushed that).